data/provider/src/main/java/tech/pegasys/teku/api/NodeDataProvider.java

```diff
@@ -89,13 +94,30 @@ public NodeDataProvider(
     this.isLivenessTrackingEnabled = isLivenessTrackingEnabled;
     this.proposersDataManager = proposersDataManager;
     this.forkChoiceNotifier = forkChoiceNotifier;
+    this.spec = spec;
   }
 
   public List<Attestation> getAttestations(
       final Optional<UInt64> maybeSlot, final Optional<UInt64> maybeCommitteeIndex) {
     return attestationPool.getAttestations(maybeSlot, maybeCommitteeIndex);
   }
 
+  public ObjectAndMetaData<List<Attestation>> getAttestationsAndMetaData(
+      final Optional<UInt64> maybeSlot, final Optional<UInt64> maybeCommitteeIndex) {
+    return lookupMetaData(
+        attestationPool.getAttestations(maybeSlot, maybeCommitteeIndex), maybeSlot);
+  }
+
+  private ObjectAndMetaData<List<Attestation>> lookupMetaData(
+      final List<Attestation> attestations, final Optional<UInt64> maybeSlot) {
+    Preconditions.checkArgument(
+        !attestations.isEmpty() || maybeSlot.isPresent(),
+        "Unable to determine spec version. No slot parameter provided and no attestations found");
```

Comment:
humm.. so the API will fail if we have no attestation and no slot as parameter. I wonder if we can default to PHASE0 in that case.
I don't like to fail in that case just because we can't decide the milestone, which is actually unused because the array will be empty.

Reply:
Good catch, ended up using the recent chain data in the end: try getting the current slot otherwise fallback to 0. The same logic we implemented when retrieving the attestations from the pool